arraycontext/impl/pyopencl.py

```diff
@@ -371,10 +325,9 @@ def transform_loopy_program(self, t_unit):
             if "i1" in all_inames:
                 inner_iname = "i1"
         else:
-            raise RuntimeError(
-                "Unable to reason what outer_iname and inner_iname "
-                f"needs to be; all_inames is given as: {all_inames}"
-            )
+            # cannot "fit" the optimization strategy for the provided kernel
+            # => bail
+            return t_unit
```

Comment:
Agreed. This was there because of the there wasn't a transformation strategy for the fused-batch kernels. But, given that that's been fixed recently, reverted back to raising.

Reply:
> fused-batch kernels.

To elaborate on this: on the meshmode end, the interpolation batch kernels had to be fused so that those operations could be supported by array-contexts that do not allow in-place modification of arrays. We had this "do-nothing" transformation strategy for those kernels as custom-iname tagging didn't become a thing until recently.

Given that that's been fixed on the meshmode end, we reverted it back to raising an error when no-transform strategy could be found.

Reply:
OIC, thanks for explaining.